pkg/apis/secretmanager/v1alpha1/externalsecret_types.go

```diff
@@ -65,7 +65,9 @@ type ObjectReference struct {
 type KeyReference struct {
 	// The key in the generated secret to place fetched secret value into.
 	SecretKey string `json:"secretKey"`
-
+	// Namespace of the secret. Ignored if SecretStore
+	// +optional
+	Namespace *string `json:"namespace,omitempty"`
```

Comment:
ExternalSecrets are namespaced scope, they should only be able to create a secret in the same namespace as the ExternalSecret.

Reply:
removed